package/MDAnalysis/analysis/dielectric.py

```diff
@@ -65,6 +65,10 @@ class DielectricConstant(AnalysisBase):
     the usual case if electrostatics are handled with a Ewald summation
     technique. See [Neumann1983]_ for details on the derivation.
 
+    .. warning::
+      Applying this class requires that no free charges, such as ions or
+      charged fragments, are present in the simulation.
```

Comment:
@PicoCentauri  perhaps I am misunderstanding but you can have no  charged anything or is it that you need overall neutrality. Apologoies if silly q.

Reply:
Nono it is fine. This is really a bit tricky. There are even publications where people do this wrong.

As we do it in the assertion you need charge neutral molecules. In a classical MD world they can have of course have partial charges. In fact, they have to because otherwise the system's total dipole moment would be zero and with it the dielectric constant.

What you can not have are ions such as Na+ Cl- or other charged fragments i.e negative carboxylic acid groups COO- or positive amine groups H3N+.